caddy/caddymain/run.go

```diff
@@ -409,13 +415,71 @@ func initTelemetry() error {
 	return nil
 }
 
+// LoadEnvFromFile loads additional envs if file provided and exists
+// Envs in file should be in KEY=VALUE format
+func LoadEnvFromFile(envFile string) error {
+	if envFile == "" {
+		return nil
+	}
+
+	file, err := os.Open(envFile)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	envMap := make(map[string]string)
+
+	scanner := bufio.NewScanner(file)
+	var line string
+	lineNumber := 0
+
+	for scanner.Scan() {
+		line = scanner.Text()
+		lineNumber++
+
+		// skip lines starting with comment
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		// skip empty line
+		if len(line) == 0 {
+			continue
+		}
+
+		fields := strings.SplitN(line, "=", 2)
+		if len(fields) != 2 {
+			return fmt.Errorf("Can't parse line %d; line should be in KEY=VALUE format", lineNumber)
+		}
+
+		key := strings.TrimSpace(fields[0])
+		val := fields[1]
+
+		if key == "" {
+			return fmt.Errorf("Can't parse line %d; KEY can't be empty string", lineNumber)
+		}
+		envMap[key] = val
+	}
```

Comment:
Check the scanner's error after the loop: `if err := scanner.Err(); err != nil { ... }`

Reply:
Oh, ok. Didn't know about scanner error, heh, but it's obvious. I will fix this